src/io/flutter/logging/FlutterLogEntryParser.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.logging;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.openapi.util.Key;
+import org.dartlang.vm.service.element.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+import static io.flutter.logging.FlutterLog.LOGGING_STREAM_ID;
+
+public class FlutterLogEntryParser {
+
+  // Known entry categories.
+  public static final String LOG_CATEGORY = "LOG";
+  public static final String DAEMON_CATEGORY = "DAEMON";
+  public static final String RAW_CATEGORY = "RAW";
+
+  @Nullable
+  public static FlutterLogEntry parse(@Nullable String id, @Nullable Event event) {
+    if (id != null && event != null) {
+      switch (id) {
+        case LOGGING_STREAM_ID:
+          return parseLoggingEvent(event);
+      }
+    }
+
+    return null;
+  }
+
+  @Nullable
+  static FlutterLogEntry parseDaemonEvent(@NotNull ProcessEvent event, @Nullable Key outputType) {
```

Comment:
My suspicion is that the daemon events aren't that interesting to app developers - that we want to give users insight into their app at runtime, not the steps that flutter_tools goes through when it launches their app.

Reply:
> Anyway, this we can certainly iterate on.

👍